assets/app/scripts/controllers/buildConfig.js

```diff
@@ -103,6 +102,12 @@ angular.module('openshiftConsole')
           updateFilterWarning();
           LabelFilter.addLabelSuggestionsFromResources($scope.unfilteredBuilds, $scope.labelSuggestions);
           LabelFilter.setLabelSuggestions($scope.labelSuggestions);
+
+          // Sort now to avoid sorting on every digest loop.
+          $scope.orderedBuilds = orderByDate($scope.builds, true);
+          if ($scope.orderedBuilds.length) {
+            $scope.latestBuild = $scope.orderedBuilds[0];
```

Comment:
You're right. I tested this and didn't see a problem because the whole latest build block is hidden by an `ng-if="(builds | hashSize) !== 0"` in the view, but I should clear it here, too.